server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSource.java

```diff
@@ -74,6 +77,16 @@ public boolean needsScores() {
         return false;
     }
 
+    /**
+     * Build a function prepares rounding values to be called many times.
+     * <p>
+     * This returns a {@linkplain Function} because auto date histogram will
+     * need to call it many times over the course of running the aggregation.
+     */
+    public Function<Rounding, Rounding.Prepared> roundingPreparer(IndexReader reader) throws IOException {
```

Comment:
After sleeping on it, I've come to think that having the default here is wrong.  I think we should either make this abstract and require each specific ValuesSource to specify if they round or not, or at a minimum make the default to throw and let ValuesSources that support rounding override it.  Otherwise, people writing new ValuesSources need to know that if they aren't going to support rounding, they need to have an override, which is counter intuitive IMHO.

Reply:
I like `abstract` pretty well. A similar thing has been bothering me that I think this'll solve. I'll ping you when I've pushed something.